### PR TITLE
Aws jobs

### PR DESCRIPTION
**Pull request by Q-lds**
The basic usage is outlined in the test file: `test_aws_managed_jobs.py`

```
input_data_path = os.path.join(
    os.environ["AMZN_BRAKET_INPUT_DIR"], "input_data/"
)

# Create the qubo and the qaoa
q = QAOA()
q.set_device(create_device('aws', 'arn:aws:braket:::device/quantum-simulator/amazon/sv1'))
q.compile(self.vc)
q.dump(file_name='openqaoa_params.json', file_path=input_data_path, prepend_id=False, overwrite=True)

job = LocalQuantumJob.create(
    device="arn:aws:braket:::device/quantum-simulator/amazon/sv1",
    source_module="./tests/jobs_test_input/aws_braket_source_module/openqaoa_qaoa_script.py",
    image_uri="amazon-braket-oq-dev",
    input_data={"input_data": input_data_path},
)
```

### CODE REVIEW

File: src/openqaoa-core/qaoa_components/ansatz_constructor/baseparams.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 from .operators import Hamiltonian
 from .hamiltonianmapper import HamiltonianMapper
+from .gatemap import RotationGateMap
```
Comment (Q-lds) [inline src/openqaoa-core/qaoa_components/ansatz_constructor/baseparams.py:8]: @vishal-ph this should fix the doc issue ... up to you if we want to find an alternative solution

File: src/openqaoa-core/utilities.py [hunk 2]
```diff
@@ -1683,20 +1683,20 @@ def dicke_wavefunction(excitations, n_qubits):
...
         Upper bound on number of excitations in the basis vector
-    n_qubits:
+    n_qubits: int
         Total number of qubits in the system
 
     Returns
     -------
     wavefunction:
-        The wavefunction vector for a given cumulative Dicke states with <=k excitations
+        The wavefunction vector for a given cumulative Dicke states with $<=k$ excitations
```
Comment (vishal-ph) [inline src/openqaoa-core/utilities.py:1699]: if this latex works than maybe we can update it to $\leq k$

File: tests/test_workflows.py [hunk 3]
```diff
@@ -825,16 +863,13 @@ def test_qaoa_asdict_dumps(self):
...
-            name="test", 
+            description="test", 
             run_by="raul", 
             provider="-", 
             target="-", 
             cloud="local", 
             client="-", 
-            qubit_routing="-", 
-            error_mitigation="-", 
-            error_correction="-"
```
Comment (shahidee44) [inline tests/test_workflows.py:836]: Double checking if these headers are planned to be removed. And is `name` to be replaced with `description`

File: src/openqaoa-core/algorithms/rqaoa/rqaoa_workflow.py [hunk 4]
```diff
@@ -51,17 +51,17 @@ class RQAOA(Workflow):
         For a complete list of its parameters and usage please see the method set_circuit_properties
     rqaoa_parameters: `RqaoaParameters`
         Set of parameters containing all the relevant information for the recursive procedure of RQAOA.
-    results: `RQAOAResults`
+    results: `RQAOAResul`
```
Comment (shahidee44) [inline src/openqaoa-core/algorithms/rqaoa/rqaoa_workflow.py:54]: Deleted the `t` in all RQAOAResult in this file :x

File: src/openqaoa-core/algorithms/rqaoa/rqaoa_result.py [hunk 5]
```diff
@@ -73,24 +73,24 @@ def from_dict(
         dictionary:dict
     ):
         """
-        Creates a RQAOAResults object from a dictionary (which is the output of the asdict method).
+        Creates a RQAOAResul object from a dictionary (which is the output of the asdict method).
```
Comment (shahidee44) [inline src/openqaoa-core/algorithms/rqaoa/rqaoa_result.py:76]: Same here

File: tests/test_aws_managed_jobs.py [hunk 6]
```diff
@@ -0,0 +1,220 @@
+#   Copyright 2022 Entropica Labs
```
Comment (shahidee44) [inline tests/test_aws_managed_jobs.py:1]: Need to remove Apache License

File: tests/test_workflows.py (hunk 3 above)
Comment (Q-lds) [reply tests/test_workflows.py:836]: Yes, that is correct!

File: src/openqaoa-core/algorithms/rqaoa/rqaoa_workflow.py (hunk 4 above)
Comment (Q-lds) [reply src/openqaoa-core/algorithms/rqaoa/rqaoa_workflow.py:54]: Fixed

File: tests/test_aws_managed_jobs.py (hunk 6 above)
Comment (Q-lds) [reply tests/test_aws_managed_jobs.py:1]: Done

File: src/openqaoa-core/algorithms/rqaoa/rqaoa_result.py (hunk 5 above)
Comment (Q-lds) [reply src/openqaoa-core/algorithms/rqaoa/rqaoa_result.py:76]: Done

**Review by shahidee44 — APPROVED** [review]

**Pull request merged**